src/commands/context/selectDockerContext.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ProgressLocation, window } from 'vscode';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../localize';
+import { dockerContextManager, IDockerContextListItem } from '../../utils/dockerContextManager';
+
+export async function selectDockerContext(prompt: string): Promise<IDockerContextListItem> {
+    const fetchingContexts = localize('vscode-docker.commands.context.fetchingContexts', 'Fetching Docker contexts...');
+    let contexts: IDockerContextListItem[];
+    await window.withProgress(
+        { location: ProgressLocation.Notification, title: fetchingContexts },
+        async () => {
+            contexts = await dockerContextManager.listAll()
+        });
+
+    const contextItems = contexts.map(ctx => ({
+        label: `${ctx.Name} ${ctx.Current ? '*' : ''}`.trim(),
+        context: ctx
```

Comment:
It's not significantly different but other places use `IAzureQuickPickItem<T>` which is designed for this sort of quickpick, in this case that would mean instead of putting the context in `context`, it would be in `data`.
```typescript
const contextItems: IAzureQuickPickItem<IDockerContextListItem>[] = ...
    {
        label: ...
        data: ctx
    }
```

And below, `return selectedContextItem.data;`

Reply:
Updated